### ISSUE TITLE
Add latest YouTube video block to home page

### ISSUE THREAD

**Issue by geerlingguy** [geerlingguy/jeffgeerling-com#137]
I'd like to have a custom block which displays a video thumbnail, then title, then date, for the three most recent YouTube videos on my channel. There's gotta be some simple API I could use to get this to automatically pull in. Set a cache on the block so it generates once every hour or so, and bob's your uncle!

**Comment by TechStudent11** [comment]
Not sure if this is too late but I found this on Stack Overflow; https://stackoverflow.com/a/23253789

All that's left is getting the latest video ID.

**Comment by minecraftchest1** [comment]
This script gets the video ID and name using yt-dlp if you are intersted. That would allow you to link directly to the video if you want. yt-dlp spits out other information that could be useful. Then the only thing would be figuring how to update the homepage with that info.

**Comment by minecraftchest1** [comment]
Yes.



Wyatt Jackson
    Matrix: @minecraftchest1:matrix.org
    Discord: Minecraftchest1#8670
    Telegram: https://t.me/minecraftchest1
    Mastodon: @***@***.***
    Github: https://github.com/minecraftchest1
    Gitlab: https://gitlab.com/minecraftchest1
    Email: ***@***.***

Sent from my T-Mobile 4G LTE Device



-------- Original message --------
From: TechStudent10 ***@***.***>
Date: 2/12/22 2:25 PM (GMT-06:00)
To: geerlingguy/jeffgeerling-com ***@***.***>
Cc: Minecraftchest1 ***@***.***>, Comment ***@***.***>
Subject: Re: [geerlingguy/jeffgeerling-com] Add latest YouTube video block to home page (Issue #137)


This script gets the video ID and name using yt-dlp if you are intersted. That would allow you to link directly to the video if you want. yt-dlp spits out other information that could be useful. Then the only thing would be figuring how to update the homepage with that info.

yt-dlp --playlist-items 1 --skip-download --dump-json  'https://www.youtube.com/c/JeffGeerling' | jq '.id, .title'


Some other fields that might be useful

"webpage_url",
"upload_date",
"view_count".
"categories[]",
"categories[]",
"like_count",
"fulltitle",
"width",
"height",
"resolution",
"thumbnail",
"description",


Isn't that a CLI tool?

**Comment by TechStudent11** [comment]
> Yes.

And this is a website

**Comment by minecraftchest1** [comment]
Yes it is a command line tool. Yes, we are talking about content on a website. However, this website we are talking about has a way to update content through some sort of API. I imagine it would not be hard to have a script run the command out of a cron job and update the page using curl. If not, a plugin could likely also be written to run the command on the server. There is also the possibility of using an iframe or something similar to embed a page that shows the video or other info using a page rendered using CGI, go webapp, python webapp, nodejs webapp, etc. The Json output could also be saved to a file accessable to the public updated by a cron job and accessed by a plugin or via client side javascript.

**Comment by TechStudent11** [comment]
> > Yes it is a command line tool. Yes, we are talking about content on a website. However, this website we are talking about has a way to update content through some sort of API. I imagine it would not be hard to have a script run the command out of a cron job and update the page using curl. If not, a plugin could likely also be written to run the command on the server. There is also the possibility of using an iframe or something similar to embed a page that shows the video or other info using a page rendered using CGI, go webapp, python webapp, nodejs webapp, etc. The Json output could also be saved to a file accessable to the public updated by a cron job and accessed by a plugin or via client side javascript.
> 
> Yeah, I guess that works

**Comment by geerlingguy** [comment]
I think I'll cancel doing this for the time being. I like having the blog be a blog, and my YouTube channel being a YouTube channel. It also avoids loading all of Google's garbage on many page loads, as long as I don't embed a video too high up in a post that appears on the home page.
